tgui/packages/tgui/interfaces/ReagentGrinder.js

```diff
@@ -0,0 +1,166 @@
+import { useBackend } from '../backend';
+import { Box, Button, Section, Table, Flex, Icon, Dimmer } from '../components';
+import { Window } from '../layouts';
+import { BeakerContents } from '../interfaces/common/BeakerContents';
+
+export const ReagentGrinder = (props, context) => {
+  return (
+    <Window resizable>
+      <Window.Content scrollable display="flex" className="Layout__content--flexColumn">
+        <Operating/>
+        <GrinderControls/>
+        <GrinderContents/>
+        <GrinderReagents/>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const Operating = (props, context) => {
+  const { data } = useBackend(context);
+  const {
+          operating,
+          name
+        } = data;
+
+  if (operating) {
+    return (
+      <Dimmer>
+        <Flex mb="30px">
+          <Flex.Item
+            bold
+            color="silver"
+            textAlign="center">
+            <Icon
+              name="spinner"
+              spin
+              size={4}
+              mb="15px"
+            /><br />
+            The {name} is processing...
+          </Flex.Item>
+        </Flex>
+      </Dimmer>
+    );
+  }
+};
```

Comment:
This is used in like 3 different places now, if you didn't mind, I would appreciate if you could make this into its own separate component and just import it into this file, all the other places that also have this

Reply:
Has been moved into its own component and have replaced the identical code used in KitchenMachines.js